src/ImmersedBoundaries/masked_reductions.jl

```diff
@@ -0,0 +1,79 @@
+using Oceananigans.Fields: AbstractField, ReducedField
+using CUDA: @allowscalar
+import Statistics: norm
+
+# ###
+# ###  reduction operations involving immersed boundary grids exclude the immersed region 
+# ### (we exclude also the values on the faces of the immersed boundary with `solid_interface`)
+# ###
+
+const ImmersedField = AbstractField{<:Any, <:Any, <:Any, <:ImmersedBoundaryGrid}
+
+struct MaskedObject{LX, LY, LZ, O, G, M, T} <: AbstractArray{T, 3} 
```

Comment:
```suggestion
struct MaskedOperation{LX, LY, LZ, O, G, M, T} <: AbstractOperation{LX, LY, LZ, G, T} 
```

note that `AbstractOperation` subtypes `AbstractArray` (via `AbstractField`).